anyway/flask_app.py

```diff
@@ -1135,15 +1139,78 @@ def load_identity_when_session_expires():
+#     endpoint=None,
+#     view_func=single_news_flash,
+#     methods=["GET"],
+# )
 app.add_url_rule("/api/news-flash", endpoint=None, view_func=news_flash, methods=["GET"])
 
 
+parser = reqparse.RequestParser()
+parser.add_argument('id', type=int, help='News flash id')
+parser.add_argument('source', type=str, help='news flash source')
+parser.add_argument('start_date', type=int, help='limit news flashes to a time period starting at the given timestamp')
+parser.add_argument('end_date', type=int, help='limit news flashes to a time period ending at the given timestamp')
+parser.add_argument('interurban_only', type=bool, help='limit news flashes to inter-urban')
+parser.add_argument('road_number', type=int, help='limit news flashes to given road')
+parser.add_argument('road_segment_only', type=bool, help='limit news flashes to items where a road_segment is specified')
+parser.add_argument('offset', type=int, help='skip items from start to given offset')
+parser.add_argument('limit', type=int, help='limit number of retrieved items to given limit')
+
+news_flash_fields_model = api.model('news_flash', {
+    'id': fields.Integer(),
+    'accident': fields.Boolean(description='This news-flash reports an accident'),
+    'author': fields.String(),
+    'date': fields.Integer(description='format: timestamp'),
+    'description': fields.String(),
+    'lat': fields.Float(),
+    'link': fields.String(),
+    'lon': fields.Float(),
+    'road1': fields.Float(),
+    'road2': fields.Float(),
+    'resolution': fields.String(description='Type of location of this news-flash'),
+    'title': fields.String(),
+    'source': fields.String(),
+    'organization': fields.String(),
+    'location': fields.String(),
+    'tweet_id': fields.Integer(),
+    'region_hebrew': fields.String(),
+    'district_hebrew': fields.String(),
+    'yishuv_name': fields.String(),
+    'street1_hebrew': fields.String(),
+    'street2_hebrew': fields.String(),
+    'non_urban_intersection_hebrew': fields.String(),
+    'road_segment_name': fields.String()
+})
+news_flash_list_model = api.model('news_flash_list', {
+    'news_flashes': fields.List(fields.Nested(news_flash_fields_model))
+})
+
+
+@api.route("/api/news-flash/<int:news_flash_id>")
+class RetrieveSingleNewsFlash(Resource):
+
+    @api.doc('get single news flash')
+    # @api.expect(parser)
+    @api.response(404, 'News flash not found')
+    @api.response(200, 'Retrieve single news-flash item', news_flash_fields_model)
+    def get(self, news_flash_id):
+        return single_news_flash(news_flash_id)
+
+
+@api.route("/api/news-flash-new", methods=["GET"])
+class RetrieveNewsFlash(Resource):
+
+    @api.doc('get news flash records')
+    @api.expect(parser)
+    @api.response(404, 'Parameter value not supported or missing')
+    @api.response(200, 'Retrieve news-flash items filtered by given parameters', news_flash_list_model)
+    def get(self):
+        return news_flash()
```

Comment:
I am not sure that i understand the concept of `@api.expect`, in the case of `RetrieveSingleNewsFlash` the `single_news_flash` function is getting params but `@api.expect(parser)` is not used and in the case of `RetrieveNewsFlash` the `news_flash` function is not getting any params but `@api.expect(parser)` is used.

Why is that?

Reply:
You are right. In the single news flash, the id is part of the URL, so it is not described in the `@api.exect` construct. In the `RetrieveNesFlash` the parameters are described in the `api.expect`. In the current implementation they are not passed to the `news_flash()` method, as it is a partial implementation. I will complete it it, and then they will be passed to the `news_flash()` method, and will be consumed directly from the parser (rather then from the `request`).
`
@api.route("/api/news-flash-new", methods=["GET"])
class RetrieveNewsFlash(Resource):

    @api.doc('get news flash records')
    @api.expect(parser)
    @api.response(404, 'Parameter value not supported or missing')
    @api.response(200, 'Retrieve news-flash items filtered by given parameters', news_flash_list_model)
    def get(self):
        args = parser.parse_args()
        return news_flash_new(args)

`